stock_valuation_fifo_lot/wizard/stock_valuation_layer_revaluation.py

```diff
@@ -0,0 +1,244 @@
+            move_lines = self._search_svl_move_lines(lot)
+            qty = sum(move_lines.mapped("qty_remaining"))
+            if qty > 0:
+                matching_lot_ids.append(lot.id)
+        return {"domain": {"lot_id": [("id", "in", matching_lot_ids)]}}
+
+    @api.depends("product_id")
+    def _compute_display_lot_id(self):
+        for rec in self:
+            rec.display_lot_id = False
+            if (
+                rec.product_id.tracking != "none"
+                and rec.product_id.cost_method == "fifo"
+            ):
+                rec.display_lot_id = True
+
+    @api.depends("lot_id")
+    def _compute_current_value_lot_svl(self):
+        for rec in self:
+            rec.current_quantity_lot_svl = 0.0
+            rec.current_value_lot_svl = 0.0
+            if not rec.lot_id:
+                continue
+            move_lines = rec._search_svl_move_lines(rec.lot_id)
+            rec.current_quantity_lot_svl = sum(move_lines.mapped("qty_remaining"))
+            rec.current_value_lot_svl = sum(move_lines.mapped("value_remaining"))
+
+    @api.depends(
+        "current_value_svl",
+        "current_quantity_svl",
+        "added_value",
+        "current_value_lot_svl",
+        "current_quantity_lot_svl",
+        "lot_id",
+    )
+    def _compute_new_value(self):
+        revaluations = self.filtered(lambda l: l.lot_id)
+        for reval in revaluations:
+            reval.new_value = reval.current_value_lot_svl + reval.added_value
+            if not float_is_zero(
+                reval.current_quantity_lot_svl,
+                precision_rounding=self.product_id.uom_id.rounding,
+            ):
+                reval.new_value_by_qty = (
+                    reval.new_value / reval.current_quantity_lot_svl
+                )
+            else:
+                reval.new_value_by_qty = 0.0
+        return super(
+            StockValuationLayerRevaluation, self - revaluations
+        )._compute_new_value()
+
+    def action_validate_revaluation(self):
+        self.ensure_one()
+        if self.lot_id:
+            # Overrides standard behavior to store the remaining value in a new SVL rather than
+            #  modifying the existing one.
+            if self.currency_id.is_zero(self.added_value):
+                raise UserError(
+                    _("The added value doesn't have any impact on the stock valuation")
+                )
+            product_id = self.product_id.with_company(self.company_id)
+            remaining_svls = self.env["stock.valuation.layer"].search(
+                [
+                    ("product_id", "=", product_id.id),
+                    ("remaining_qty", ">", 0),
+                    ("company_id", "=", self.company_id.id),
+                    ("lot_ids", "in", self.lot_id.id),
+                ]
+            )
+            remaining_svls = remaining_svls.filtered(
+                lambda l: l._get_unconsumed_in_move_line(self.lot_id)
+            )
```

Comment:
I think we need to switch to using the `stock.move.line` records rather than SVLs when it comes to identifying the remaining qty and value. The current implementation doesn't block you from devaluing more than the selected lot contains.